dcc-submission-loader/src/main/java/org/icgc/dcc/submission/loader/core/SubmissionLoader.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2016 The Ontario Institute for Cancer Research. All rights reserved.                             
+ *                                                                                                               
+ * This program and the accompanying materials are made available under the terms of the GNU Public License v3.0.
+ * You should have received a copy of the GNU General Public License along with                                  
+ * this program. If not, see <http://www.gnu.org/licenses/>.                                                     
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.submission.loader.core;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.icgc.dcc.common.core.util.stream.Collectors.toImmutableList;
+import static org.icgc.dcc.common.core.util.stream.Collectors.toImmutableMap;
+import static org.icgc.dcc.submission.loader.core.DependencyFactory.createLoadFilesResolver;
+import static org.icgc.dcc.submission.loader.file.ReleaseFilesLoaderFactory.createReleaseFilesLoader;
+import static org.icgc.dcc.submission.loader.util.Releases.getReleases;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import lombok.Cleanup;
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+import org.icgc.dcc.submission.loader.cli.ClientOptions;
+import org.icgc.dcc.submission.loader.model.FileTypePath;
+import org.icgc.dcc.submission.loader.model.Project;
+
+import com.google.common.base.Stopwatch;
+
+@Slf4j
+public class SubmissionLoader {
+
+  public static void loadSubmission(ClientOptions options) throws IOException {
```

Comment:
`@NonNull`. 

Also, I wouldn't pass CLI layer classes to lower layers. This causes dependency cycles. There is a case to be made for being DRY, so I see why you chose to do this. No good solution for that I guess.